lua/grapple-utils.lua

```diff
@@ -12,4 +13,59 @@ function M.move_to_end()
 	end
 end
 
+local function get_current_index(tags, current_path)
+	for i, tag in ipairs(tags) do
+		if tag.path == current_path then
+			return i
+		end
+	end
+	return nil
+end
+
+local function swap_tags(tags, index1, index2)
+	tags[index1], tags[index2] = tags[index2], tags[index1]
+	for i = 1, #tags do
+		grapple.untag({ path = tags[i].path })
+		grapple.tag({ path = tags[i].path })
+	end
+end
+
+local function move_tag(direction)
+	local current_path = vim.api.nvim_buf_get_name(0)
+	local current_opts = { path = current_path }
+	local tags, err = grapple.tags()
+
+	if not tags then
+		vim.notify("No tags available: " .. err, vim.log.levels.WARN)
+		return
+	end
+
+	if grapple.exists(current_opts) then
```

Comment:
Might be clearer to use a guard clause here like the one above.

Reply:
And the same thing with the ones below. I think its better because it keeps the error messages close to the checks and removes a lot of nesting.